examples/darkmode/src/components/ThemeManager.astro

```diff
@@ -0,0 +1,69 @@
+---
+type Props = {
+  "default-theme"?: "auto" | "dark" | "light";
+};
+
+const { "default-theme": defaultTheme = "auto" } = Astro.props;
+---
+
+<theme-manager data-default-theme={defaultTheme}></theme-manager>
+<script is:inline>
+  if (!customElements.get("theme-manager")) {
+    class ThemeManager extends HTMLElement {
+      connectedCallback() {
+        this.initThemeManagement(this.dataset.defaultTheme);
+      }
+
+      initThemeManagement(defaultTheme) {
+        window.theme = (() => {
```

Comment:
Are we sure it will only be initialized once?

Reply:
yes due to the whole thing being wrapped in `if (!customElements.get("theme-manager"))`